core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -307,23 +309,31 @@ public void removeWorkflow(String workflowId, boolean archiveWorkflow) throws Ex
 	}
 
 	public SearchResult<WorkflowSummary> search(String query, String freeText, int start, int size, List<String> sortOptions) {
-		
+
 		SearchResult<String> result = indexer.searchWorkflows(query, freeText, start, size, sortOptions);
 		List<WorkflowSummary> workflows = result.getResults().stream().parallel().map(workflowId -> {
 			try {
-				
+
 				WorkflowSummary summary = new WorkflowSummary(executionDAO.getWorkflow(workflowId, false));
 				return summary;
-				
+
 			} catch(Exception e) {
 				logger.error(e.getMessage(), e);
 				return null;
 			}
 		}).filter(Objects::nonNull).collect(Collectors.toList());
-		int missing = result.getResults().size() - workflows.size();
+
+		//Search again to filter out some workflows that might have been in flux state before.
+		//It's possible that during the first search ES had some workflows that were already completed in dynamo but
+		//still had a 'RUNNING' state in ES
+		SearchResult<String> repeatedResult = indexer.searchWorkflows(query, freeText, start, size, sortOptions);
```

Comment:
Thinking more this, search it repeated may not solve the problem. Let's revisit the approach.